daemon/cmd/daemon_main.go

```diff
@@ -1261,6 +1264,10 @@ func initEnv(logger *slog.Logger, vp *viper.Viper) {
 		logger.Warn("IPSec encrypted overlay is enabled but IPSec is not. Ignoring option.")
 	}
 
+	if option.Config.EnableRemoteNodeMasquerade && !option.Config.EnableBPFMasquerade {
+		logging.Fatal("EnableRemoteNodeMasquerade is enabled but EnableBPFMasquerade is not. Ignoring option.")
```

Comment:
If this is fatal, then the options are not being ignored. This is the common pattern but should be expressed more clearly to the user. Also needs to be rebased for `slog` migration in the upstream tree:
```suggestion
		logging.Fatal(logger, "Option " + option.EnableRemoteNodeMasquerade + " requires BPF masquerade to be enabled (" + option.EnableBPFMasquerade + ")")
```

Reply:
I rebased it already and made the recommended changes